cds-snc/covid-alert-app
# Feature/otk notification

Part 3 - `OTK notification`

Follow up PRs to continue the rest of the full as outlined below.


## Full Description
10% of people who have entered a OTK have not uploaded their codes.

## Motivation
When someone has entered their OTK but has not uploaded their codes
They need to know they are not finished their task of letting people know they've been exposed
And they need to know what to do next
So that their close contacts will receive an exposure notification


## Acceptance criteria

**Main screen**

Given people have entered their OTK but have not uploaded their codes
When they leave the flow or otherwise land on the main page of the app
Then they see messaging that they are not quite done ([design](https://www.figma.com/file/0pLawavG1fnBTXSYn4pdKZ/Exposure-notification?node-id=14333%3A26))

**Menu**

Given people have entered their OTK but have not uploaded their codes
When they open the menu
Then they see the prompt to share exposures ([design](https://www.figma.com/file/0pLawavG1fnBTXSYn4pdKZ/Exposure-notification?node-id=14333%3A52))

**Intermediate instructions screen**

Given people have entered their OTK but have not uploaded their codes
When they click "Finish sharing exposures"
Then they see the intermediate screen with steps, and the first step is now "Done" ([design](https://www.figma.com/file/0pLawavG1fnBTXSYn4pdKZ/Exposure-notification?node-id=14333%3A740))
And when they click "Next"
Then they follow the rest of the existing process (entering date of symptom onset, etc)

**OTK Notification**
If people have only entered the OTK but have not shared, they will receive a notification reminding them to complete it.
(https://www.figma.com/file/0pLawavG1fnBTXSYn4pdKZ/Exposure-notification?node-id=14333%3A74)

**Confirmation of completed the flow**

When people have entered their OTK and shared their exposures, completing the flow
Then their screen says "Thanks for helping fight COVID-19" and tells them to share any new exposures going forward ([text change - design)](https://www.figma.com/file/0pLawavG1fnBTXSYn4pdKZ/Exposure-notification?node-id=14333%3A521)

**Sharing exposures on subsequent days**
Given someone has entered a OTK and shared exposures on a previous day
When they open their app
Then they see the "Share new exposures screen" ([text change - design](https://www.figma.com/file/0pLawavG1fnBTXSYn4pdKZ/Exposure-notification?node-id=14333%3A1112)).

------------------------------------------------------------------------------------------------------------------------
This PR includes the push notification that users will get if they close the app or turn off the screen before they've completed uploading their OTK. 

This PR also includes some UI changes for the DiagnosedView and DiagnosedShareView screens.


Testing OTK Notification:

- Install COVID Alert App
- Onboard and select a province
- Enter OTK
- Press the home button to put the app in the background, the notification banner will appear on the top
- If you turn off the screen, notification will display on the lock screen
